L1Trigger/L1TCaloLayer1/src/UCTCard.hh

```diff
@@ -0,0 +1,68 @@
+#ifndef UCTCard_hh
+#define UCTCard_hh
+
+#include <vector>
+
+#include "UCTGeometry.hh"
+
+class UCTRegion;
+
+class UCTCard {
+public:
+
+  UCTCard(uint32_t crt, uint32_t crd);
+
+  virtual ~UCTCard();
+
+  // To set up event data before processing
+
+  const std::vector<UCTRegion*>& getRegions() const {return regions;}
+  const UCTRegion* getRegion(uint32_t rgn) const {return regions[rgn];}
+  const UCTRegion* getRegion(UCTRegionIndex r) const;
+
+  // To process event
+
+  bool clearEvent();
+  bool setECALData(UCTTowerIndex t, bool ecalFG, uint32_t ecalET);
+  bool setHCALData(UCTTowerIndex t, uint32_t hcalFB, uint32_t hcalET);
+  bool process();
+
+  // More access functions
+
+  const uint32_t getCrate() const {return crate;}
+  const uint32_t getCard() const {return card;}
+
+  const uint32_t et() const {return cardSummary;}
+
+  friend std::ostream& operator<<(std::ostream&, const UCTCard&);
+
+private:
+
+  // No default constructor is needed
+
+  UCTCard();
+
+  // No copy constructor is needed
+
+  UCTCard(const UCTCard&);
+
+  // No equality operator is needed
+
+  const UCTCard& operator=(const UCTCard&);
+
+  // Helper functions
+
+  const UCTRegion* getRegion(bool negativeEta, uint32_t caloEta, uint32_t caloPhi) const;
+
+  // Owned card level data 
+
+  uint32_t crate;
+  uint32_t card;
+
+  std::vector<UCTRegion*> regions;
```

Comment:
Most of the comments made are appropriate and I have no problem fixing them as suggested.  The reason for many of the decisions, e.g. use of #define instead of namespace/const value, is because of my being old :(  However, the code works now, and would like to request that we make a release with existing stuff with minimal compliance modifications, while a fully compliant version is provided for the next update by Nick and company who are half my age!

We don't want to copy regions - we need to set the data of existing UCTRegion objects and similarly for all other objects created to represent the hardware. I am sure there are other ways of accomplishing this than through pointers, but this is what I chose.  I would imagine that  making this change will take longer to debug, so I suggest postponing to a later update.